library/psa_crypto_rsa.c

```diff
@@ -274,7 +274,10 @@ static psa_status_t rsa_export_public_key(
 #endif /* defined(BUILTIN_KEY_TYPE_RSA_KEY_PAIR) ||
         * defined(BUILTIN_KEY_TYPE_RSA_PUBLIC_KEY) */
 
-#if defined(BUILTIN_KEY_TYPE_RSA_KEY_PAIR)
+/* XXX Does MBEDTLS_PSA_ACCEL_KEY_TYPE_RSA_KEY_PAIR use these functions? Both
+ * of them or just rsa_generate_key? */
+#if defined(BUILTIN_KEY_TYPE_RSA_KEY_PAIR) && \
+    defined(MBEDTLS_GENPRIME)
 static psa_status_t psa_rsa_read_exponent( const uint8_t *domain_parameters,
```

Comment:
You need to change the guard where this function is used in `psa_crypto.c` as well.

Reply:
Added, although without fully seeing how accelerators related to the software implementation, I'm not sure it's correct. My worry is we will return "sorry, not supported" when the accelerator does provide a way to generate an RSA key and `MBEDTLS_GENPRIME` is unset. I would expect for the most part that `MBEDTLS_GENPRIME` will be set even when generators are used, given how the `PSA_WANT_*` options work (there is no way for an accelerator to say "don't provide a software implementation for RSA key generation").

Reply:
I think we are OK if MBEDTLS_GENPRIME is always set where we want to use accelerators and want to generate RSA keys.

Reply:
Yes, let's just say that for now if your RSA accelerator supports key generation then you must set `MBEDTLS_GENPRIME`.